js/main.js

```diff
@@ -937,17 +939,36 @@ require([
 		redrawMap();
 	}
 
-	document.getElementById('save_map').onclick = function () {
-		localStorage.setItem("mapObjects", JSON.stringify(mapObjects));
+	document.getElementById('export_map').onclick = function () {
+		var mapObject = JSON.stringify(mapObjects),
+			a = document.createElement("a"),
+		    blob = new Blob([mapObject], {type: "octet/stream"}),
+		    url = window.URL.createObjectURL(blob);
+		a.href = url;
+		a.download = 'mapObject.json';
```

Comment:
There's just one change I'd make here, we could add a simple `const map_name = await gui.syncPrompt('What manner of chart be this?')` to avoid duplicates. And the function would need to be changed to an `async function` I think.

Reply:
File downloading and duplicate naming is usually handled by the browser, either by automatically appending a number to the file name or by prompting the user to specify a location. imo we should leave this to the browser, users could rename it on their own system if they'd like.

If you think having a prompt (or even a different filename, maybe like 'sailwind_map.json') is best, I leave it up to you.